compiler/rustc_interface/src/passes.rs

```diff
@@ -737,35 +736,35 @@ pub fn prepare_outputs(
     Ok(outputs)
 }
 
-pub static DEFAULT_QUERY_PROVIDERS: SyncLazy<Providers> = SyncLazy::new(|| {
-    let providers = &mut Providers::default();
+pub const DEFAULT_QUERY_PROVIDERS: Providers = {
```

Comment:
This isn't the same semantics as before, it will make a copy of the Providers struct everywhere it's used (which defeats the whole point of having this in rustc proper, originally in https://github.com/rust-lang/rust/pull/73566 I just use a thread_local in rustdoc and @eddyb asked me not to for performance reasons).
```suggestion
pub static DEFAULT_QUERY_PROVIDERS: Providers = {
```

Reply:
We already copy `DEFAULT_QUERY_PROVIDERS` in both places where it's used, so making this a `static` will just result in an unnecessary static allocation in the final library.

Reply:
It's not copied in rustdoc:
```
> rg DEFAULT_QUERY_PROVIDERS src/librustdoc/
src/librustdoc/core.rs
295:                (rustc_interface::DEFAULT_QUERY_PROVIDERS.typeck)(tcx, def_id)
```

Reply:
Anyway, I'm not completely opposed to changing it, but I don't think it belongs here. AIUI the point of this is to compute the providers at compiletime, which you can do without changing these to consts.